### ISSUE TITLE
UnenclosedJsonRecordReader fails to read escaped strings

### ISSUE THREAD

**Issue by lfrancke** [Esri/spatial-framework-for-hadoop#68]
`UnenclosedJsonRecordReader` fails to read JSON that includes escaped characters which leads to all kinds of errors (like trying to read a whole input file at once)

Example:

````
{
  "foo": "foo\"bar"
}
````

I have a fix ready and I'm happy to submit a pull request but it changes a whole bunch of code related to other things as well so I'm not sure when I get around to it: https://github.com/lfrancke/spatial-framework-for-hadoop/blob/fixes/json/src/main/java/com/esri/json/hadoop/UnenclosedJsonRecordReader.java

**Comment by lfrancke** [comment]
@randallwhitman Sorry for the long silence and thank you for working on this.

Yes, I have a free week coming up and I'll do my best to get it reviewed until next Thursday.

**Comment by randallwhitman** [comment]
@lfrancke Sounds good.

**Comment by randallwhitman** [comment]
Yes, quite a few edge cases overall.  It seemed for a while that every time I tested one thing, I thought of something else to test, and found another edge case.

In `moveToRecordStart` the edge case is: getting an opening brace when expecting either "attributes" or "geometry", which indicates the previously-found opening brace from which we thought we were starting, must have been inside a string literal, after which the double-quote we found, must have been a closing quote mark rather than the opening quote mark.  (The other changes are handling backslash as escape, and simplifying with the `getChar` methods.)

But also, the change might be more understandable by looking at the associated test case - it is tested by `TestEscOpenLast` with `unenclosed-json-esc5.json`.

Thanks for your feedback.
